specs/disk_details_for_hardware_inventory.adoc

```diff
@@ -0,0 +1,190 @@
+= Specification for adding disk details in node hardware inventory
+
+Add disk details also as a part of node hardware inventory
+
+== Problem description
+
+Hardware inventory details are used to show exact status of the particular
+node. Currently disk details are missing in node hardware inventory. Add all possible
+disk details in node hardware inventory and store it in etcd.
+
+== Use Cases
+
+Tendrl should show the current status of disk.
+
+== Proposed change
+* From node agent collect all possible disk details in node inventory list and store
+  it in etcd. (Detailed information in implementation section).
+
+== Alternatives
+
+* hdparm is also giving some disk details like serial number and firmware but it
+  works only for ATA disks types. For SCSI disk we have to use sdparm.
+
+== Data model impact
+
+* Add used disks id in etcd with key nodes/node_id/Disks/Used.
+* Add free disks id in etcd with key nodes/node_id/Disks/Free.
+* Add all disks details in etcd with key nodes/node_id/Disks.
+* There needs to be one object per disk, which contains all the information about
+  the disk. This needs to be in the disks directory. The object's key needs to be
+  the id that's guaranteed to always be available.
+* The objects under used and free directories under disks, need to be empty
+  objects with their key being the id of the disk object under disks.
+```
+     list: nodes/$Node_context.node_id/Disks
+     list: nodes/$Node_context.node_id/Disks/Used
+     list: nodes/$Node_context.node_id/Disks/Free
+```
+
+== Impacted modules
+
+== Tendrl API impact:
+
+None
+
+=== Tendrl/common impact
+
+None
+
+=== Tendrl/node_agent impact
+
+* Add a new function in pull_hardware_inventory to collect disk details.
+* Create three different classes Disks, FreeDisks, UsedDisks in persistence
+  module.
+* Modify the manager to retrieve inventory details from tendrl-node-inventory
+  file and compare it with node_inventory list, If any changes then populate the
+  disk details in class instance and store it in etcd.
+* Add three different object in tendrl definitions for disks, used disks and
+  free disks.
+* Commands used to retrieve disk details are:
+```
+1. columns = 'NAME,FSTYPE,MOUNTPOINT,UUID,SIZE,ROTA
+   lsblk --all --bytes --noheadings --output='%s' --path --raw" % columns
+2. hwinfo --block --only disk_name
```

Comment:
no i am not @nthomas-redhat told me no need to store the data in file and parse it directly